pkg/openwsn/contrib/radio_hal.c

```diff
@@ -0,0 +1,316 @@
+
+openwsn_radio_t openwsn_radio;
+
+/* stores the event capture time */
+static PORT_TIMER_WIDTH _txrx_event_capture_time = 0;
+
+void _idmanager_addr_override(void)
+{
+    /* Initiate Id manager here and not in `openstack_init` function to
+       allow overriding the short id address before additional stack
+       components are initiated */
+    idmanager_init();
+
+    /* override 16b address to avoid short address collision */
+    network_uint16_t short_addr;
+    luid_get_short(&short_addr);
+    open_addr_t id;
+    id.type = ADDR_16B;
+    memcpy(&id.addr_16b, short_addr.u8, IEEE802154_SHORT_ADDRESS_LEN);
+    idmanager_setMyID(&id);
+
+    /* override PANID */
+    id.type = ADDR_PANID;
+    network_uint16_t panid_be = byteorder_htons(OPENWSN_PANID);
+    memcpy(&id.addr_16b, &panid_be, IEEE802154_SHORT_ADDRESS_LEN);
+    idmanager_setMyID(&id);
+
+    /* recover ADDR_64B */
+    eui64_t eui64;
+    eui64_get(eui64.uint8);
+
+    /* Set all IEEE addresses */
+    uint16_t panid = OPENWSN_PANID;
+    ieee802154_radio_set_hw_addr_filter(openwsn_radio.dev, &short_addr,
+                                        &eui64, &panid);
+
+}
+
+static void _hal_radio_cb(ieee802154_dev_t *dev, ieee802154_trx_ev_t status)
+{
+    (void)dev;
+
+    debugpins_isr_set();
+    _txrx_event_capture_time = sctimer_readCounter();
+    debugpins_isr_clr();
+
+    switch (status) {
+    case IEEE802154_RADIO_CONFIRM_TX_DONE:
+        ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
+                                               IEEE802154_TRX_STATE_TRX_OFF);
+        openwsn_radio.endFrame_cb(_txrx_event_capture_time);
+        break;
+    case IEEE802154_RADIO_INDICATION_RX_DONE:
+        openwsn_radio.endFrame_cb(_txrx_event_capture_time);
+        break;
+    case IEEE802154_RADIO_INDICATION_TX_START:
+        openwsn_radio.startFrame_cb(_txrx_event_capture_time);
+        break;
+    case IEEE802154_RADIO_INDICATION_RX_START:
+        openwsn_radio.startFrame_cb(_txrx_event_capture_time);
+        break;
+    default:
+        break;
+    }
+}
+
+int openwsn_radio_init(void *radio_dev)
+{
+    assert(radio_dev);
+    ieee802154_dev_t *dev = (ieee802154_dev_t *)radio_dev;
+
+    LOG_DEBUG("[openwsn/radio]: initialize riot-adaptation\n");
+    openwsn_radio.dev = dev;
+
+    /* override short_address and panid*/
+    _idmanager_addr_override();
+
+    if (ieee802154_radio_request_on(dev)) {
+        LOG_ERROR("[openwsn/radio]: unable to initialize device\n");
+        return -1;
+    }
+
+    /* Set the Event Notification */
+    dev->cb = _hal_radio_cb;
+
+    /* If the radio is still not in TRX_OFF state, spin */
+    while (ieee802154_radio_confirm_on(dev) == -EAGAIN) {}
+
+    /* Enable basic mode, no AUTOACK. no CSMA */
+    ieee802154_radio_set_rx_mode(dev, IEEE802154_RX_AACK_DISABLED);
+    ieee802154_radio_set_csma_params(dev, NULL, -1);
+
+    /* MAC layer will handle retransmissions */
+    if (ieee802154_radio_has_frame_retrans(dev)) {
```

Comment:
From the Radio HAL API, `set_csma_params`:
```c
     * @param[in] retries number of CSMA-CA retries. If @p retries < 0,
     *                    retransmissions with CSMA-CA MUST be disabled.
     *                    If @p retries == 0, the @ref
     *                    ieee802154_radio_ops::request_transmit function is
     *                    equivalent to CCA send.

```

This means `set_csma_params` will already disable retransmissions. The reason is because hardware retransmission software don't support "retransmissions without CSMA-CA". Therefore this lines can be removed

Reply:
removed